app/controllers/qsets_controller.rb

```diff
@@ -10,6 +10,36 @@ def index
     end
   end
 
+  def qset_json
+    params.permit(:filter)
+    filter = params[:filter]
+    answers_json_data = []
+    $questionCount = 0
+    if filter == 'other'
+      @raw_questions = Question.includes(:answers).where(qset_id: @qset.id).where.not(user_id: current_user)
+      @raw_questions.each do |raw_question|
+        raw_question_id = raw_question.id
+        raw_answer = Answer.where(question_id: raw_question_id)
+        answers_json_data.push(raw_answer)
+      end
+    elsif filter =='mine'
+      @raw_questions = Question.includes(:answers).where(qset_id: @qset.id, user_id: current_user)
+      @raw_questions.each do |raw_question|
+        raw_question_id = raw_question.id
+        raw_answer = Answer.where(question_id: raw_question_id)
+        answers_json_data.push(raw_answer)
+      end
+    else
+      @raw_questions = Question.includes(:answers).where(qset_id: @qset.id)
+      @raw_questions.each do |raw_question|
+        raw_question_id = raw_question.id
+        raw_answer = Answer.where(question_id: raw_question_id)
+        answers_json_data.push(raw_answer)
+      end
+    end
+    render :json => {:questions => @raw_questions, :answers => answers_json_data}
```

Comment:
Given answers are indeed nested relations of questions, we can make the JSON objects returned by our API reflect that, instead of requiring the caller to know we're sending back a separate set of objects. I've incorporated that into the suggested refactored code.